ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d.cpp

```diff
@@ -35,8 +43,13 @@ Tensor MaxPool2DOp::invoke(uint8_t queue_id, const Tensor& input_tensor, uint32_
 
     if (!memory_config.shard_spec.has_value()) {
         // Input is not sharded. Perform sharding.
+        applied_shard_scheme = applied_shard_scheme == TensorMemoryLayout::INTERLEAVED ? TensorMemoryLayout::HEIGHT_SHARDED : applied_shard_scheme; // default to height sharding
```

Comment:
maybe you can try picking the best choice here based on tensor dims.
In our torch ttn integration, compiler will just pass in interleaved tensor and it's up to the op to do the "right" thing.
We can do this as a next step as well.